model/training/data/converter.py

```diff
@@ -0,0 +1,138 @@
+import json
+from cv2 import imread, IMREAD_GRAYSCALE
+
+
+class DataConverter:
+    def __init__(self):
+        self.path = "/home/aneta/Pulpit/Studia/dp-proj/proj/model/training/data/"
+        self.training_loaded = 0
+        self.test_loaded = 0
+        self.val_loaded = 0
+        self.images_entries = []
+        self.annotations_entries = []
+
+    def convert_data_to_COCO(self):
+
+        path = self.path + "image_lists/" + "images_list.txt"
+        ann_id = 1
+        img_id = 1
+
+        with open(path) as file:
+            for position, line in enumerate(file):
+                image_name = line.strip()
+                img = self.load_image(image_name)
+
+                im_entry = {
+                    'file_name': image_name,
+                    'height': img.shape[0],
+                    'width': img.shape[1],
+                    'id': img_id
+                }
+
+                self.images_entries.append(im_entry)
+
+                ann_id = self.load_annotation(image_name, img.shape, ann_id, img_id)
+                img_id += 1
+
+                if img_id % 1000 == 0:
+                    print(img_id)
+
+        json_en = {
+            "images": self.images_entries,
+            "annotations": self.annotations_entries,
+            "categories": [
+                {
+                    "id": 1,
+                    "name": "table",
+                    "supercategory": "none"
+                },
+                {
+                    "id": 2,
+                    "name": "cell",
+                    "supercategory": "none"
+                }
+            ]
+        }
+        json_obj = json.dumps(json_en, indent=2)
+        with open("coco.json", "w") as outfile:
+            outfile.write(json_obj)
+
+    def load_image(self, img_name: str):
+        img_file_name = self.path + "images/" + img_name
+        return imread(img_file_name, IMREAD_GRAYSCALE)
+
+    def area(self, bbox):
+        return (bbox[2] - bbox[0]) * (bbox[3] - bbox[1])
+
+    def segmentation(self, bbox):
+        return [bbox[0], bbox[1], bbox[0], bbox[3], bbox[2], bbox[3], bbox[2], bbox[1]]
```

Comment:
Tutaj potrzebujemy jeszcze jeden poziom zagłębienia. To powinna być lista 2D, która wewnątrz ma jedną listę z tymi wartościami